kernel/aster-nix/src/process/signal/pauser.rs

```diff
@@ -113,9 +116,12 @@ impl Pauser {
             let sig_mask = {
                 let current_thread = current_thread!();
                 let poxis_thread = current_thread.as_posix_thread().unwrap();
-                let mut current_sigmask = *poxis_thread.sig_mask().lock();
+                // Add self.sig_mask to sig_mask of current_thread
+                let mut current_sigmask = poxis_thread.sig_mask().lock();
+                self.old_sig_mask
+                    .store(current_sigmask.as_u64(), Ordering::Relaxed);
```

Comment:
I'm not sure if it's reasonable to add the field `old_sig_mask` within `Pauser` to record and restore the sigmask. If this field is shared between processes, could it be modified by another process before restoration, resulting in an incorrect mask being restored? If it's not shared between processes, then could this variable be a temporary variable within the function instead of being a field of `Pauser`?

Reply:
> I'm not sure if it's reasonable to add the field `old_sig_mask` within `Pauser` to record and restore the sigmask.

I have the same concern. @StevenJiang1110 Any response to this question?

Reply:
The problem is fixed.

After reviewing the code, I found the `is_interrupted` flag should also not be shared among pausing threads. So `is_interrupted` should also be made a local variable. The problem is also fixed.